subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/DependencyManagementResultsAsInputsIntegrationTest.groovy

```diff
@@ -148,8 +181,30 @@ class DependencyManagementResultsAsInputsIntegrationTest extends AbstractHttpDep
         fails "verify"
 
         then:
-        failureDescriptionStartsWith("Execution failed for task ':verify'.")
-        failureHasCause("Cannot fingerprint input property 'input'")
+        failureDescriptionStartsWith("Some problems were found with the configuration of task ':verify' (type 'TaskWithInput').")
+        failureDescriptionContains("Type 'TaskWithInput' property 'direct' has $annotation annotation used on property of type 'ResolvedArtifactResult'.")
+        failureDescriptionContains("Type 'TaskWithInput' property 'providerInput' has $annotation annotation used on property of type 'Provider<ResolvedArtifactResult>'.")
+        failureDescriptionContains("Type 'TaskWithInput' property 'propertyInput' has $annotation annotation used on property of type 'Property<ResolvedArtifactResult>'.")
+        failureDescriptionContains("Type 'TaskWithInput' property 'setPropertyInput' has $annotation annotation used on property of type 'SetProperty<ResolvedArtifactResult>'.")
+        failureDescriptionContains("Type 'TaskWithInput' property 'listPropertyInput' has $annotation annotation used on property of type 'ListProperty<ResolvedArtifactResult>'.")
+        failureDescriptionContains("Type 'TaskWithInput' property 'mapPropertyInput' has $annotation annotation used on property of type 'MapProperty<String, ResolvedArtifactResult>'.")
+        failureDescriptionContains("Type 'TaskWithInput' property 'nestedInput.nested' has $annotation annotation used on property of type 'Property<ResolvedArtifactResult>'.")
+
+        // Because
+        failureDescriptionContains("ResolvedArtifactResult is not supported on task properties annotated with $annotation.")
+
+        // Possible solutions
+        failureDescriptionContains("1. Extract artifact metadata and annotate with @Input.")
+        failureDescriptionContains("2. Extract artifact files and annotate with @InputFiles.")
+
+        // Documentation
+        failureDescriptionContains("Please refer to https://docs.gradle.org/${GradleVersion.current().version}/userguide/validation_problems.html#unsupported_value_type for more details about this problem.")
+
+        where:
+        annotation    | description
+        "@Input"      | "at-input"
```

Comment:
Is it bad to include an `@` in the unrolled test name? I'd just use the annotation directly.